### PR TITLE
Support `AsyncEngine` to SQLAlchemy

### LINKED ISSUE #713: async sqlalchemy engine support
### Description

It looks like in `logfire[aiohttp,fastapi,sqlalchemy]>=2.9.0`, the `sqlalchemy` `AsyncEngine` is not supported. Is it possible to add support for this? 

## Code
```python
POSTGRES: Final[AsyncEngine] = create_async_engine(url=POSTGRES_URL, echo=SETTINGS.DEBUG)
logfire.instrument_sqlalchemy(engine=POSTGRES)
```
## Error
```
Argument of type "AsyncEngine" cannot be assigned to parameter "engine" of type "Engine | None" in function "instrument_sqlalchemy"
  Type "AsyncEngine" is not assignable to type "Engine | None"
    "AsyncEngine" is not assignable to "Engine"
    "AsyncEngine" is not assignable to "None"
```

### PR DESCRIPTION
**Pull request by Kludex**
- Closes #713

### CODE REVIEW

File: tests/otel_integrations/test_sqlalchemy.py [hunk 1]
```diff
@@ -207,6 +208,182 @@ class AuthRecord(Base):
...
+                    'logfire.msg': 'DELETE FROM auth_records WHERE auth_records.id = ?',
+                    'db.statement': 'DELETE FROM auth_records WHERE auth_records.id = ?',
+                    'db.system': 'sqlite',
+                    'db.name': 'example.db',
+                },
+            },
+        ]
+    )
+
+    SQLAlchemyInstrumentor().uninstrument()
```
Comment (alexmojaki) [inline tests/otel_integrations/test_sqlalchemy.py:384]:
```suggestion
```

File: tests/otel_integrations/test_sqlalchemy.py [hunk 2]
```diff
@@ -207,6 +208,182 @@ class AuthRecord(Base):
...
+    try:
+        yield engine
+    finally:
+        path.unlink()
+
+
+@pytest.mark.anyio
+async def test_sqlalchemy_async_instrumentation(exporter: TestExporter):
+    with sqlite_async_engine(Path('example.db')) as engine:
+        logfire.instrument_sqlalchemy(engine=engine)
```
Comment (alexmojaki) [inline tests/otel_integrations/test_sqlalchemy.py:224]:
```suggestion
        logfire.instrument_sqlalchemy(engine)
```

**Review by alexmojaki — APPROVED** [review]
Great, thanks!

File: tests/otel_integrations/test_sqlalchemy.py (hunk 1 above)
Comment (Kludex) [reply tests/otel_integrations/test_sqlalchemy.py:384]: We can't, bug in OTel regarding uninstrumentation/instrumentation.

**Pull request merged**